src/api_midlevel.jl

```diff
@@ -185,3 +185,7 @@ function write_chunk(dataset_id, index::Integer, buf::Union{DenseArray,Base.Fast
     offset = [reverse(get_chunk_offset(dataset_id, index))...]
     write_chunk(dataset_id, offset, buf; dxpl_id = dxpl_id, filter_mask = filter_mask)
 end
+
+# Accept a Filter subtype as an argument
+API.h5z_get_filter_info(::Type{F}) where F <: Filters.Filter = API.h5z_get_filter_info(Filters.filterid(F))
+API.h5z_filter_avail(::Type{F}) where F <: Filters.Filter = API.h5z_filter_avail(Filters.filterid(F))

```

Comment:
These shouldn't overload the `API` function. Mid-level functions define their own function name, internally call `API`  methods,    have the `h5.._` prefix stripped, and sometimes include additional name normalization. 

One option is
```julia
get_filter_info(::Type{F}) where {F <: Filters.Filter} = API.h5z_get_filter_info(Filters.filterid(F))
get_filter_info(filter) = h5z_get_filter_info(filter)  # calls  api/helpers.jl method

filter_available(::Type{F}) where {F <: Filters.Filter} = API.h5z_filter_avail(Filters.filterid(F))
```

We might also want to include some brief documentation if they live in `midlevel` since these routines are sometimes part of the non-exported public API.

Reply:
I was originally thinking of them as helpers, but the imports were cleaner.

I was also thinking of moving this into the `Filters` module. Any thoughts about that?

Reply:
> I was also thinking of moving this into the Filters module. Any thoughts about that?

I don't have a strong opinion on this. 
On one hand, logically, it makes sense, especially if we are going to rename it.  

I'll leave the discretion up to you.